### DISCUSSION TITLE
Can not access dynamic ref inside $watch with $nextTick

### DISCUSSION THREAD

**Discussion by itsfaqih** [alpinejs/alpine#2298] [1. Help] [answered]
So I have something like this
```html
<template x-for="item in items">
    <a :x-ref="`menu-${item.id}`" x-text="item.label" :href="item.href" @mouseenter="active = item.id"></a>
</template>
```
```js
init() {
    this.$nextTick(() => {
        this.$watch('active', (value) => {
            if (value !== null) {
                this.$refs[`menu-${value}`].focus();
            }
        });
    });
}
```
But when I hover the menu item, it gives me this error.

![image](https://user-images.githubusercontent.com/48067039/139827856-7e875fdd-650d-4cbc-8bdd-8aa23d1fffe6.png)

Which means that the $watch run before it renders, or am I missing something?
I can assure you that it works if I don't use dynamic refs by the way, but I really need it to works using dynamic value because adding the id manually is time consuming.

**Comment by SimoTod** [comment, marked as answer]
You can't have dynamic ref in Alpine v3. See https://github.com/alpinejs/alpine/discussions/1527#discussioncomment-996994
You'll need to use something different, for example
```html
<template x-for="item in items">
    <a :id="`menu-${item.id}`" x-text="item.label" :href="item.href" @mouseenter="active = item.id"></a>
</template>
```
```js
init() {
    this.$nextTick(() => {
        this.$watch('active', (value) => {
            if (value !== null) {
                document.getElementById(`menu-${value}`).focus();
            }
        });
    });
}
```

**Reply by itsfaqih** [reply]
Ahh I see, I guess I'll just use what you suggested.
Thank you very much!
